pennylane/_device.py

```diff
@@ -733,16 +733,17 @@ def batch_transform(self, circuit):
             for obs in circuit.observables
             if obs.name == "Hamiltonian"
         )
+        # device property present in braket plugin
+        use_grouping = getattr(self, "use_grouping", True)
 
         hamiltonian_in_obs = "Hamiltonian" in [obs.name for obs in circuit.observables]
 
         return_types = [m.return_type for m in circuit.observables]
 
         is_shadow = ShadowExpval in return_types
+        hamiltonian_usable = not supports_hamiltonian or (finite_shots and not is_shadow)
```

Comment:
Maybe `hamiltonain_unusable`? If the device doesn't support hamiltonians, then the hamiltonian will *not* be usable by the device.

Reply:
@kshyatt-aws I'd like to clarify the name of the explanatory variable, but then I'm good to approve.